### PR TITLE
fix: add pvr pvcr commands

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?
Add package version report and package version create report commands
### What issues does this PR fix or reference?
@W-11444248@
@W-11444281@

### CODE REVIEW

File: src/commands/force/package/beta/version/create/report.ts [hunk 1]
```diff
@@ -6,17 +6,22 @@
...
+const pvclMessages = Messages.loadMessages('@salesforce/plugin-packaging', 'package_version_create_list');
+const plMessages = Messages.loadMessages('@salesforce/plugin-packaging', 'package_list');
 
 export class PackageVersionCreateReportCommand extends SfdxCommand {
   public static readonly description = messages.getMessage('cliDescription');
   public static readonly longDescription = messages.getMessage('cliLongDescription');
   public static readonly help = messages.getMessage('help');
   public static readonly;
-  public static readonly orgType = SfdxPropertyKeys.DEFAULT_DEV_HUB_USERNAME;
+  public static readonly orgType = OrgConfigProperties.TARGET_DEV_HUB;
```
Comment (WillieRuemmele) [inline src/commands/force/package/beta/version/create/report.ts:24]: I've been deleting this from commands, I don't think it's used for anything. same for pvr

File: src/commands/force/package/beta/version/report.ts [hunk 2]
```diff
@@ -30,9 +42,190 @@ export class PackageVersionReportCommand extends SfdxCommand {
...
+  private massageResultsForDisplay(results: PackageVersionReportResult): PackageVersionReportResultModified {
+    const record = JSON.parse(JSON.stringify(results)) as PackageVersionReportResultModified;
+    record.Version = [record.MajorVersion, record.MinorVersion, record.PatchVersion, record.BuildNumber].join('.');
+
+    if (results.PackageType !== 'Managed') {
+      record.AncestorVersion = 'N/A';
+      record.AncestorId = 'N/A';
+    }
+
+    if (results.Package2.IsOrgDependent === true || results.ValidationSkipped === true) {
```
Comment (WillieRuemmele) [inline src/commands/force/package/beta/version/report.ts:201]: I'm sure we can omit all of these `=== true` should be inferred from the type

File: src/commands/force/package/beta/version/report.ts [hunk 3]
```diff
@@ -6,16 +6,28 @@
...
+import { PackageVersion, PackageVersionReportResult, PackagingSObjects } from '@salesforce/packaging';
+import * as chalk from 'chalk';
 
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/plugin-packaging', 'package_version_report');
-
+const pvlMessages = Messages.loadMessages('@salesforce/plugin-packaging', 'package_version_list');
+const plMessages = Messages.loadMessages('@salesforce/plugin-packaging', 'package_list');
+export type PackageVersionReportResultModified = Omit<
+  PackageVersionReportResult,
```
Comment (WillieRuemmele) [inline src/commands/force/package/beta/version/report.ts:18]: are these types wrong in the lib?

**Review by WillieRuemmele — CHANGES_REQUESTED** [review]

File: src/commands/force/package/beta/version/report.ts (hunk 3 above)
Comment (peternhale) [reply src/commands/force/package/beta/version/report.ts:18]: @WillieRuemmele the type in library is correct. The report command's JSON result is heavily modified. In order to maintain backward compatibility and to leave a type in the library that properly represents the actual types, I had to make this new type that matches expected JSON result. Ugly, ugly. This is something that can be addressed when packaging commands land in sf.

**Review by WillieRuemmele — APPROVED** [review]

**Review by shetzel — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
### QA Notes
---
`package:beta:version:report`
human output in table
```
 Description                    null               
 Branch                         weeek              
 Tag                            null  
```
vs new
```
 Description                                       
 Branch                         weeek              
 Tag                                 
```

and present with `--json`
old:
```
    "Description": null,
    "Tag": null,
```
vs new
```
    "Description": " ",
    "Tag": " ",
```

**Comment by peternhale** [comment]
> ### QA Notes
> `--help` missing examples `package:beta:version:report` human output in table
> 
> ```
>  Description                    null               
>  Branch                         weeek              
>  Tag                            null  
> ```
> 
> vs new
> 
> ```
>  Description                                       
>  Branch                         weeek              
>  Tag                                 
> ```
> 
> and present with `--json` old:
> 
> ```
>     "Description": null,
>     "Tag": null,
> ```
> 
> vs new
> 
> ```
>     "Description": " ",
>     "Tag": " ",
> ```
> 
> `version:create:report` `--help` missing examples human output in table old
> 
> ```
>  Package Id                    0Ho3i000000Gmj6CAC 
>  Package Version Id            null               
>  Subscriber Package Version Id null               
>  Tag                           null  
> ```
> 
> vs new
> 
> ```
>  Package Id                    0Ho3i000000Gmj6CAC 
>  Package Version Id                               
>  Subscriber Package Version Id                    
>  Tag                                
> ```
> 
> same `--json` bug as above

Logged bug to address these.

**Pull request merged**